server/src/internalClusterTest/java/org/elasticsearch/indices/DataStreamIT.java

```diff
@@ -464,6 +470,33 @@ public void testAliasActionsFailOnDataStreamBackingIndices() throws Exception {
             "support aliases."));
     }
 
+    public void testNoTimestampInDocument() throws Exception {
```

Comment:
We have good integration test coverage, but no unit tests. It would be great to add a unit test at the level of the mapping code that checks the document validation. Perhaps `DateFieldMapperTests` would be a good place for this?

Reply:
Ignore this comment, the unit tests are missing because this is a 'draft' :)